code/modules/hydroponics/hydroponics_chemreact.dm

```diff
@@ -7,9 +7,9 @@
 	if(myseed)
 		myseed.on_chem_reaction(reagents) //In case seeds have some special interactions with special chems, currently only used by vines
 	//FINE WE'LL DO IT THIS WAY.
-	for(var/c in reagents.reagent_list)
-		var/datum/reagent/chem = c
-		chem.on_hydroponics_apply(myseed, reagents, src, user)
+	for(var/datum/reagent/chem in reagents.reagent_list)
+		if(chem)
+			chem.on_hydroponics_apply(myseed, reagents, src, user)
```

Comment:
what's this meant to do? If you don't have the chem it shouldn't be in the reagent list.

Also this should still be untyped unless you're expecting non-reagents to be here (it shouldn't be)

Reply:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#istypeless-for-loops

Reply:
I've legitimately never seen this before, good to know.